packages/frontend/src/features/metricFlow/components/MetricFlowSqlCard.tsx

```diff
@@ -25,14 +25,15 @@ const MetricFlowSqlCard: FC<Props> = ({
     canRedirectToSqlRunner = false,
 }) => {
     const [isOpen, setIsOpen] = useState<boolean>(false);
-    const sqlRunnerUrl = useMemo(() => {
-        const params = new URLSearchParams({
-            sql_runner: JSON.stringify({
-                sql: sql ?? '',
-            }),
-        });
-        return `/projects/${projectUuid}/sqlRunner?${params.toString()}`;
-    }, [projectUuid, sql]);
+
+    const sqlRunnerLink = useMemo(
+        () => ({
+            pathname: `/projects/${projectUuid}/sql-runner`,
+            state: { sql: sql ?? '' }, // pass SQL as state
```

Comment:
```suggestion
            state: { sql: data }
```

We can just pass `data` and not an empty string to be more explicit